### PR TITLE
fix: reapply security fix and docs for v0.2

### PR DESCRIPTION
**Pull request by arunsanna**
## Summary

Reapplication of PRs #19 and #28 to the v0.2 branch architecture.

### Changes

**1. Security Fix: Validate Before Sanitize** (from PR #19)
- Swaps the order of validation and `html.escape()` sanitization in `/generate` endpoint
- Prevents potential bypass attacks where malicious input could be transformed into something that passes validation

**2. CONTRIBUTING.md** (from PR #28, updated for v0.2)
- Development setup instructions (Docker + local)
- Branch naming and PR workflow conventions
- Updated project architecture diagram reflecting v0.2 structure
- Testing instructions
- Community links including weekly call info

## Analysis of Original PRs

| PR | Description | Status in v0.2 |
|----|-------------|----------------|
| #18 | PURL encoding fix | ✅ Already in v0.2 |
| #19 | Validate before sanitize | 🔧 **Reapplied here** |
| #20 | Unit test infrastructure | ✅ Already in v0.2 |
| #21 | Structured logging | ✅ Already in v0.2 |
| #22 | Schema validation | ✅ Already in v0.2 |
| #28 | CONTRIBUTING.md | 🔧 **Reapplied here** (updated) |

## Test Plan

- [x] Validation tests pass (`pytest tests/test_validation.py`)
- [ ] Manual test of `/generate` endpoint with edge case inputs
- [ ] Review CONTRIBUTING.md for accuracy with current structure

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+pip install -r requirements.txt
+
+# Run all tests
+pytest
+
+# Run with coverage
+pytest --cov=src --cov-report=html
+
+# Run specific test file
+pytest tests/test_validation.py -v
```
Comment (afogel) [inline CONTRIBUTING.md]: tests in 0.2 are now (sadly for me) in unittest -- I know pytest can run unittest files natively, but I figure it's worth flagging

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+# Or use CLI
+python -m src.cli --model_id "microsoft/DialoGPT-medium"
+```
+
+## Areas Welcoming Contributions
+
+We especially welcome contributions in these areas:
+
+- **SPDX 3.1 export** - Add support for SPDX AI Profile format
```
Comment (afogel) [inline CONTRIBUTING.md]: 🔥

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+   ```
+3. **Set up your environment**:
+
+   ```bash
+   # Using Docker (recommended)
+   docker build -t aibom .
+   docker run -p 7860:7860 aibom
+
+   # Or local Python setup
+   pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (eaglei15) [reply CONTRIBUTING.md]: Good idea, would you be able to adjust?
